alicloud/resource_alicloud_emr_cluster.go

```diff
@@ -0,0 +1,456 @@
+package alicloud
+
+import (
+	"github.com/aliyun/alibaba-cloud-sdk-go/sdk/requests"
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/emr"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-alicloud/alicloud/connectivity"
+)
+
+func resourceAlicloudEmrCluster() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAlicloudEmrClusterCreate,
+		Read:   resourceAlicloudEmrClusterRead,
+		Update: resourceAlicloudEmrClusterUpdate,
+		Delete: resourceAlicloudEmrClusterDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"cluster_type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"emr_ver": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"charge_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"host_group": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				ForceNew: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"host_group_name": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"host_group_type": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"charge_type": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"period": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"node_count": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"instance_type": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"disk_type": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"disk_capacity": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"disk_count": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"sys_disk_type": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"sys_disk_capacity": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"auto_renew": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
```

Comment:
> Please set validateFunc for `auto_renew` and I think it should be `bool`

fixed

Reply:
> Please set validateFunc for `auto_renew` and I think it should be `bool`

Because of the bug of SDK Generator，the Go SDK convert the boolean to string